Formula/gomigrate.rb

```diff
@@ -0,0 +1,26 @@
+class Gomigrate < Formula
+  desc "Database migrations CLI tool"
+  homepage "https://github.com/golang-migrate/migrate"
+  url "https://github.com/golang-migrate/migrate/archive/v3.3.0.tar.gz"
+  sha256 "59ade052631d6a064d3ac71df5f03c120c3029210d8d13972b6c5db7558c490f"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/golang-migrate/migrate").install buildpath.children
+
+    # Build and install cli as "migrate"
+    cd "src/github.com/golang-migrate/migrate" do
+      system "dep", "ensure", "-vendor-only"
+      system "make", "build-cli", "VERSION=v#{version}"
```

Comment:
The make target should only build for macOS.

Reply:
does temporary build directory got deleted after installation? makefile does not provide single target build, maintaining another build procedure is expensive.

Reply:
The vast majority of Makefiles for Go projects handle this correctly. It is a trivial request. See for instance

https://github.com/istio/fortio/blob/master/Makefile
https://github.com/pilosa/pilosa/blob/master/Makefile

Reply:
It is trivial in upstream, your comment sounds like we are going to do this in formulae

Reply:
Sorry for not being clear. Yes, I mean that we need a change to the upstream Makefile.